leader-election-impl/src/test/java/com/palantir/paxos/SqlitePaxosStateLogTest.java

```diff
@@ -102,4 +127,39 @@ private static PaxosValue valueForRound(long round) {
         ThreadLocalRandom.current().nextBytes(bytes);
         return new PaxosValue("someLeader", round, bytes);
     }
+
+    private static Supplier<Connection> createReusableMemoizedConnection() {
+        Supplier<Connection> baseConnectionSupplier = SqliteConnections.createDatabaseForTest();
+        Supplier<Connection> nonClosingConnectionSupplier = bypassCloseOnConnection(baseConnectionSupplier);
+        return Suppliers.memoize(nonClosingConnectionSupplier::get);
+    }
+
+    private static Supplier<Connection> bypassCloseOnConnection(Supplier<Connection> connectionSupplier) {
+        return Suppliers.compose(conn ->
+                        (Connection) Proxy.newProxyInstance(Connection.class.getClassLoader(),
+                                new Class<?>[] {Connection.class},
+                                new CloseIgnoringInvocationHandler(conn)),
+                connectionSupplier::get);
+    }
+
+    /**
+     * JDBI closes the connection after executing a query. This is desired behaviour, but does not play nicely with in
```

Comment:
Is there no way to amend this behavior? How about SQLiteDataSource, does that do it better? And then also SQLiteConnectionPoolDataSource,

Reply:
Or how about we use the disk in tests too? Shouldn't really be a problem I suspect? We can give it temp directories

Reply:
OK, we use disk